src/gt4py/backend/dace_backend.py

```diff
@@ -452,16 +467,25 @@ class DaCeComputationCodegen:
     )
 
     def generate_tmp_allocs(self, sdfg):
-        fmt = "dace_handle.{name} = allocate(allocator, gt::meta::lazy::id<{dtype}>(), {size})();"
-        return [
-            fmt.format(
-                name=f"__{array_sdfg.sdfg_id}_{name}",
+        global_fmt = "dace_handle.__{sdfg_id}_{name} = allocate(allocator, gt::meta::lazy::id<{dtype}>(), {size})();"
+        threadlocal_fmt = """dace_handle.__{sdfg_id}_{name} = allocate(allocator, gt::meta::lazy::id<{dtype}>(), omp_max_threads * ({size}))();
+#pragma omp parallel
+{{
+{name} = dace_handle.__{sdfg_id}_{name} + omp_get_thread_num() * ({size});
+}}"""
+        res = [
+            (
+                global_fmt if array.storage != dace.StorageType.CPU_ThreadLocal else threadlocal_fmt
+            ).format(
+                name=name,
+                sdfg_id=array_sdfg.sdfg_id,
                 dtype=array.dtype.ctype,
                 size=array.total_size,
             )
             for array_sdfg, name, array in sdfg.arrays_recursive()
             if array.transient and array.lifetime == dace.AllocationLifetime.Persistent
```

Comment:
I get what you're trying to do here, and it's neat, but it's also a really long statement. Consider splitting this up for readability?